### PR TITLE
Moving common dependencies of the tools to a `internal` package and convert it to a module

### PR DESCRIPTION
**Pull request by ArcturusZhang**
This PR creates `tools/internal` module consisting of the packages that are widely used by some tools we are having, including
- `tools/apidiff/delta`
- `tools/apidiff/exports`
- `tools/apidiff/report`
- `tools/apidiff/markdown`
- `tools/apidiff/repo`
- `tools/internal/dirs`
- `tools/internal/modinfo`
- `tools/internal/ioext`

The following tools are removed:
- `picker` is removed in this PR, because it is no longer used and it is referencing `apidiff` as a dependency.
- `versioner` is removed in this PR, we will remain the track 1 SDK un-moduled and do not introduce major version sub-directory in track 2, therefore we no longer need its functionality any more

This PR also creates module for every tool in the `tools` directory.

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Please do not merge this PR because we still do not resolve the ambiguous import issue now.

File: tools/common/go.mod
Comment (ArcturusZhang) [reply tools/common/go.mod:1]: Well... I am using `common` instead of `internal` to make sure that some external tools will not have to rewrite everything again. But since we will be bringing the release tool in our tools, I believe we could change this to internal...

Comment (ArcturusZhang) [reply tools/common/go.mod]: I have changed `common` to `internal`

**Comment by ArcturusZhang** [comment]
Do you think we need to create CI for every tool after we change tools into modules? @jhendrixMSFT

**Comment by jhendrixMSFT** [comment]
It would be ideal to create a pipeline just for tools so that you don't have to run track 1 CI which takes forever.

**Comment by ArcturusZhang** [comment]
> It would be ideal to create a pipeline just for tools so that you don't have to run track 1 CI which takes forever.

Since it might not be easy after a bit rounds of trying, I decide we could introduce that later after some offline discussion.

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT could you please take a look again?
I just switched to [this version on my fork](https://github.com/ArcturusZhang/azure-sdk-for-go/releases/tag/v54.2.1), and everything is working now.

After this PR gets merged, the steps should be 
1. release a new version for track 1 SDK v54.2.1
2. I will open another PR to switch the azure-sdk-for-go dependency in internal to official, and release the first module version of `tools/internal` (maybe `tools/internal/v0.1.0`)
3. I will open some more PRs, each for one tool, switch the internal and azure-sdk-for-go version to v0.1.0 and the official v54.2.1

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**
